src/pages/pupils/programmes/[programmeSlug]/units/[unitSlug]/lessons/[lessonSlug]/index.tsx

```diff
@@ -177,18 +184,35 @@ export const getStaticProps: GetStaticProps<
         };
       }
 
-      let transcriptSentences = curriculumData.transcriptSentences;
-
-      if (curriculumData.videoTitle && !curriculumData.isLegacyLicense) {
-        // For new content we need to fetch the captions file from gCloud and parse the result to generate
-        // the transcript sentences.
-        const fileName = `${curriculumData.videoTitle}.vtt`;
-        transcriptSentences = (await getCaptionsFromFile(fileName)) ?? [];
-      }
+      // For new content we need to fetch the captions file from gCloud and parse the result to generate
+      // the transcript sentences.
+      const resolveTranscriptSentences = (() => {
+        if (curriculumData.videoTitle && !curriculumData.isLegacy) {
+          return getCaptionsFromFile(`${curriculumData.videoTitle}.vtt`);
+        }
+
+        return curriculumData.transcriptSentences;
+      })();
+
+      // Resolve the requests for the transcript and worksheet existence in parallel
+      const [transcriptSentences, downloadExistence] = await Promise.all([
+        resolveTranscriptSentences,
+        getDownloadResourcesExistence(
```

Comment:
Pinching this from the teacher's section — it makes a request to the downloads API (either the new or legacy one) to tell use if there is a worksheet for the lesson